WeakAurasOptions/CommonOptions.lua

```diff
@@ -1123,9 +1131,38 @@ local function PositionOptions(id, data, _, hideWidthHeight, disableSelfPoint, g
       order = 72,
       image = function() return "", 0, 0 end,
       hidden = function()
-        return IsParentDynamicGroup() or not (data.anchorFrameType == "SCREEN" or data.anchorFrameType == "UIPARENT" or data.anchorFrameType == "MOUSE" or IsGroupByFrame())
+        return
+          IsParentDynamicGroup() or
+          not (
+            data.anchorFrameType == "SCREEN" or
+            data.anchorFrameType == "UIPARENT" or
+            data.anchorFrameType == "MOUSE" or
+            data.anchorFrameType == "ACTION_BUTTON_SPELL" or
+            IsGroupByFrame()
+          )
       end,
     },
+    anchorFrameActionButtonSpell = {
+      type = "input",
+      width = WeakAuras.doubleWidth,
+      name = L["Spell"],
+      order = 72,
+      set = function(info, value)
+        if (WeakAuras.IsSpellKnownIncludingPet(value)) then
+          data.anchorFrameActionButtonSpellId = select(7, GetSpellInfo(value))
+          WeakAuras.Add(data)
+        end
+      end,
+      get = function()
+        return GetSpellInfo(data.anchorFrameActionButtonSpellId)
+      end,
+      hidden = function()
+        if (IsParentDynamicGroup() or IsGroupByFrame()) then
+          return true
+        end
+        return not (data.anchorFrameType == "ACTION_BUTTON_SPELL")
+      end
+    },
```

Comment:
Largely copied from the others. I tried to follow through the code in places to see if there was a `type = "spell"` or other properties to lean into here but I didn't have a ton of success there.